pkgs/by-name/ni/nixos-render-docs-redirects/src/nixos_render_docs_redirects/__init__.py

```diff
@@ -0,0 +1,134 @@
+import argparse
+import json
+import sys
+from pathlib import Path
+
+
+def add_content(redirects: dict[str, list[str]], identifier: str, path: str) -> dict[str, list[str]]:
+    if identifier in redirects:
+        raise IdentifierExists(identifier)
+
+    # Insert the new identifier in alphabetical order
+    new_redirects = list(redirects.items())
+    insertion_index = 0
+    for i, (key, _) in enumerate(new_redirects):
+        if identifier > key:
+            insertion_index = i + 1
+        else:
+            break
```

Comment:
I'd like to see a test on that, because I don't think the equality test in the current ones will take order into account.

Reply:
The equality test do take order into account. Comparing normal `dict` objects don't, but I've converted the `dict`s into a list of tuples, which do assert the order of elements.

Reply:
Ah right, had overlooked that!